src/libtiled/hexagonalrenderer.cpp

```diff
@@ -289,6 +318,9 @@ void HexagonalRenderer::drawTileLayer(QPainter *painter,
 
     CellRenderer renderer(painter, CellRenderer::HexagonalCells);
 
+    const int endX = map()->infinite() ? layer->bounds().width() : layer->width();
```

Comment:
No, the `- layer->x()` part is to change the bounds to local coordinates, since the `bounds()` include the layer's position. Subtracting `layer->bounds().x()` would defeat the point of using `layer->bounds().right()`.